pkg/monitoring/monitoring.go

```diff
@@ -246,14 +248,62 @@ func (tr *templateRegexp) translate(value string) *templateRegexpResult {
 func GetOverwroteAppAnswers(annotations map[string]string) map[string]string {
 	overwritingAppAnswers := annotations[cattleOverwriteAppAnswersAnnotationKey]
 	if len(overwritingAppAnswers) != 0 {
-		var appOverwriteInput mgmtv3.MonitoringInput
+		var appOverwriteInput map[string]string
 		err := json.Unmarshal([]byte(overwritingAppAnswers), &appOverwriteInput)
 		if err == nil {
-			return appOverwriteInput.Answers
+			return appOverwriteInput
 		}
 
 		logrus.Errorf("failed to parse app overwrite input from %q, %v", overwritingAppAnswers, err)
 	}
 
 	return map[string]string{}
 }
+
+func GetMonitoringCreatorID(anno map[string]string) string {
+	if anno == nil {
+		return ""
+	}
+	creator, ok := GetOverwroteAppAnswers(anno)[cattleCreatorIDAnnotationKey]
+	if ok {
+		return creator
+	}
+	return anno[cattleCreatorIDAnnotationKey]
+}
+
+func ValidateCreator(anno map[string]string, userLister mgmtv3.UserLister) (bool, error) {
+	if anno == nil {
+		return false, nil
+	}
+	creatorID := GetMonitoringCreatorID(anno)
+	user, err := userLister.Get("", creatorID)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	if user.Enabled == nil || !*user.Enabled {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func UpdateMonitoringCreator(obj metav1.Object, userMgr user.Manager, apicontext *types.APIContext) error {
+	anno := obj.GetAnnotations()
+	if anno == nil {
+		anno = map[string]string{}
+	}
+	userID := userMgr.GetUser(apicontext)
+	answers := GetOverwroteAppAnswers(anno)
+	answers[cattleCreatorIDAnnotationKey] = userID
```

Comment:
Can we store the `userID` into another annotation item? Because the answers will be set into App spec, this may cause some misunderstanding.